sample_3.py

```diff
@@ -0,0 +1,5 @@
+class Solution:
+    @lru_cache
+    def fib(self, n: int) -> int:
+        return self.fib(n-1)+self.fib(n-2) if n>1 else n
+
```

Comment:
if n would be negative then it would return negative value which is wrong. So, you need to write separate statements for n==0 and n==1. refer below code
class Solution:
    def fib(self, n: int) -> int:
        if isinstance(n, int):
            
            if n <= 0:
                return 0
     
            elif n == 1:
                return 1
     
            else:
                return self.fib(n-1)+self.fib(n-2)

        else:
            print("n should be of Integer type")

Reply:
Thanks! Added input type checking and handling of negative cases.